app/controllers/staff/base_controller.rb

```diff
@@ -1,12 +1,22 @@
+class UserNotAuthorised < StandardError; end
 class Staff::BaseController < ApplicationController
   include Secured
 
+  rescue_from(UserNotAuthorised) do
+    render "pages/errors/not_authorised", status: 401
+  end
+
   helper_method :current_user
   def current_user
-    @current_user ||= User.find_by!(identifier: signed_in_user_identifier) { |user|
-      user.name = session.dig(:userinfo, "info", "name")
-      user.email = session.dig(:userinfo, "info", "email")
-    }
+    @current_user ||= if session.dig(:userinfo)
+      User.find_by!(identifier: signed_in_user_identifier) do |user|
+        user.name = session.dig(:userinfo, "info", "name")
+        user.email = session.dig(:userinfo, "info", "email")
+      end
+    end
+  rescue ActiveRecord::RecordNotFound
+    repudiate!
```

Comment:
Repudiating the session isn't strictly required as a bad session will always cause an error in this method. It will however allow the before action in Secured to trigger earlier, and remove the need to look up the database a second time. We want each authentication attempt to have a fresh session and a clean slate to pick up any updates made in between.